AppCenter/AppCenter/Internals/Device/MSDeviceTracker.m

```diff
@@ -111,9 +111,40 @@ - (MSDevice *)device {
  */
 - (MSDevice *)updatedDevice {
   @synchronized(self) {
-    MSDevice *newDevice = [[MSDevice alloc] init];
+    MSDevice *newDevice = [MSDevice new];
 #if TARGET_OS_IOS
-    CTCarrier *carrier = [[[CTTelephonyNetworkInfo alloc] init] subscriberCellularProvider];
+    CTTelephonyNetworkInfo *telephonyNetworkInfo = [CTTelephonyNetworkInfo new];
+    CTCarrier *carrier;
+
+// TODO Use @available API and availability attribute when deprecating Xcode 8 then we can try removing these pragma.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wundeclared-selector"
+   SEL serviceSubscriberCellularProviders = @selector(serviceSubscriberCellularProviders);
+#pragma clang diagnostic pop
+    if ([telephonyNetworkInfo respondsToSelector:serviceSubscriberCellularProviders]) {
+        
+      // Call serviceSubscriberCellularProviders.
+      NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:
+                                  [telephonyNetworkInfo methodSignatureForSelector:serviceSubscriberCellularProviders]];
+      [invocation setSelector:serviceSubscriberCellularProviders];
+      [invocation setTarget:telephonyNetworkInfo];
+      [invocation invoke];
+      void *returnValue;
+      [invocation getReturnValue:&returnValue];
+      NSDictionary<NSString *, CTCarrier *> *carriers = (__bridge NSDictionary<NSString *, CTCarrier *> *)returnValue;
+      for (NSString *key in carriers) {
+        carrier = carriers[key];
+        break;
+      }
+    }
+    
+    // Use the old API as fallback if new one isn't work.
```

Comment:
`isn't` -> `doesn't`

Reply:
@jaelim-ms blaming yourself and going to rest